crates/forge-runner/src/build_trace_data/test_program_path.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Context;
+use anyhow::Result;
+use cairo_lang_sierra::program::VersionedProgram;
+use camino::Utf8PathBuf;
+use fs4::FileExt;
+use std::fs;
+use std::fs::File;
+use std::io::BufWriter;
+
+pub const TESTS_PROGRAMS_DIR: &str = ".snfoundry_test_code";
+
+#[derive(Clone)]
+pub struct TestProgramPath(Utf8PathBuf);
+
+impl TestProgramPath {
+    pub fn save_test_program(
+        versioned_program: &VersionedProgram,
+        test_program_path: Utf8PathBuf,
```

Comment:
```suggestion
        test_program_path: Utf8PathBuf,
```
Please name it differently, `source_test_program_path` maybe?

Reply:
I don't see how the word `source` is beneficial here, but okay

Reply:
Changed the logic here